# Analiza skupa podataka Hotel Booking

U ovom projektu analiziramo skup podataka **Hotel Booking**, koji sadrži informacije o rezervacijama u gradskim i resort hotelima. Cilj rada je da se kroz obradu i analizu podataka utvrdi koji faktori mogu uticati na otkazivanje rezervacije.

Kao ciljna promenljiva koristi se kolona **is_canceled**, koja označava da li je rezervacija otkazana ili nije. Problem koji rešavamo predstavlja binarnu klasifikaciju, jer model treba da predvidi jednu od dve moguće klase: rezervacija je otkazana ili rezervacija nije otkazana.

U toku rada biće urađeni sledeći koraci: učitavanje i pregled podataka, analiza nedostajućih vrednosti, eksplorativna analiza podataka, vizuelizacija odnosa između promenljivih, dodavanje novih promenljivih kroz feature engineering, priprema podataka za modelovanje i testiranje više algoritama mašinskog učenja.

## 1. Učitavanje biblioteka

Na početku učitavamo biblioteke koje su potrebne za rad sa podacima i osnovnu analizu dataset-a.

In [24]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sb

pd.set_option("display.max_columns", None)

## 2. Učitavanje skupa podataka

Dataset se učitava iz foldera **data**. Koristi se CSV fajl **hotel_booking.csv**, koji sadrži podatke o hotelskim rezervacijama.

In [25]:
df = pd.read_csv("../data/hotel_booking.csv")

print("Podaci su uspešno učitani.")
print(f"Broj redova: {df.shape[0]}")
print(f"Broj kolona: {df.shape[1]}")

Podaci su uspešno učitani.
Broj redova: 119390
Broj kolona: 36


## 3. Osnovni pregled podataka

U ovom delu proveravamo osnovnu strukturu dataset-a: prvih nekoliko redova, nazive kolona, tipove podataka i osnovne informacije o podacima.

In [26]:
df.head()

,hotel,is_canceled,lead_time,arrival_date_year,arrival_date_month,arrival_date_week_number,arrival_date_day_of_month,stays_in_weekend_nights,stays_in_week_nights,adults,children,babies,meal,country,market_segment,distribution_channel,is_repeated_guest,previous_cancellations,previous_bookings_not_canceled,reserved_room_type,assigned_room_type,booking_changes,deposit_type,agent,company,days_in_waiting_list,customer_type,adr,required_car_parking_spaces,total_of_special_requests,reservation_status,reservation_status_date,name,email,phone-number,credit_card
0,Resort Hotel,0,342,2015,July,27,1,0,0,2,0.0,0,BB,PRT,Direct,Direct,0,0,0,C,C,3,No Deposit,NaN,NaN,0,Transient,0.0,0,0,Check-Out,2015-07-01,Ernest Barnes,Ernest.Barnes31@outlook.com,669-792-1661,************4322
1,Resort Hotel,0,737,2015,July,27,1,0,0,2,0.0,0,BB,PRT,Direct,Direct,0,0,0,C,C,4,No Deposit,NaN,NaN,0,Transient,0.0,0,0,Check-Out,2015-07-01,Andrea Baker,Andrea_Baker94@aol.com,858-637-6955,************9157
2,Resort Hotel,0,7,2015,July,27,1,0,1,1,0.0,0,BB,GBR,Direct,Direct,0,0,0,A,C,0,No Deposit,NaN,NaN,0,Transient,75.0,0,0,Check-Out,2015-07-02,Rebecca Parker,Rebecca_Parker@comcast.net,652-885-2745,************3734
3,Resort Hotel,0,13,2015,July,27,1,0,1,1,0.0,0,BB,GBR,Corporate,Corporate,0,0,0,A,A,0,No Deposit,304.0,NaN,0,Transient,75.0,0,0,Check-Out,2015-07-02,Laura Murray,Laura_M@gmail.com,364-656-8427,************5677
4,Resort Hotel,0,14,2015,July,27,1,0,2,2,0.0,0,BB,GBR,Online TA,TA/TO,0,0,0,A,A,0,No Deposit,240.0,NaN,0,Transient,98.0,0,1,Check-Out,2015-07-03,Linda Hines,LHines@verizon.com,713-226-5883,************5498


In [27]:
print(df.columns.tolist())

['hotel', 'is_canceled', 'lead_time', 'arrival_date_year', 'arrival_date_month', 'arrival_date_week_number', 'arrival_date_day_of_month', 'stays_in_weekend_nights', 'stays_in_week_nights', 'adults', 'children', 'babies', 'meal', 'country', 'market_segment', 'distribution_channel', 'is_repeated_guest', 'previous_cancellations', 'previous_bookings_not_canceled', 'reserved_room_type', 'assigned_room_type', 'booking_changes', 'deposit_type', 'agent', 'company', 'days_in_waiting_list', 'customer_type', 'adr', 'required_car_parking_spaces', 'total_of_special_requests', 'reservation_status', 'reservation_status_date', 'name', 'email', 'phone-number', 'credit_card']


In [28]:
df.dtypes

hotel                                 str
is_canceled                         int64
lead_time                           int64
arrival_date_year                   int64
arrival_date_month                    str
arrival_date_week_number            int64
arrival_date_day_of_month           int64
stays_in_weekend_nights             int64
stays_in_week_nights                int64
adults                              int64
children                          float64
babies                              int64
meal                                  str
country                               str
market_segment                        str
distribution_channel                  str
is_repeated_guest                   int64
previous_cancellations              int64
previous_bookings_not_canceled      int64
reserved_room_type                    str
assigned_room_type                    str
booking_changes                     int64
deposit_type                          str
agent                             

In [29]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 119390 entries, 0 to 119389
Data columns (total 36 columns):
 #   Column                          Non-Null Count   Dtype  
---  ------                          --------------   -----  
 0   hotel                           119390 non-null  str    
 1   is_canceled                     119390 non-null  int64  
 2   lead_time                       119390 non-null  int64  
 3   arrival_date_year               119390 non-null  int64  
 4   arrival_date_month              119390 non-null  str    
 5   arrival_date_week_number        119390 non-null  int64  
 6   arrival_date_day_of_month       119390 non-null  int64  
 7   stays_in_weekend_nights         119390 non-null  int64  
 8   stays_in_week_nights            119390 non-null  int64  
 9   adults                          119390 non-null  int64  
 10  children                        119386 non-null  float64
 11  babies                          119390 non-null  int64  
 12  meal                       

In [30]:
df.describe()

,is_canceled,lead_time,arrival_date_year,arrival_date_week_number,arrival_date_day_of_month,stays_in_weekend_nights,stays_in_week_nights,adults,children,babies,is_repeated_guest,previous_cancellations,previous_bookings_not_canceled,booking_changes,agent,company,days_in_waiting_list,adr,required_car_parking_spaces,total_of_special_requests
count,119390.000000,119390.000000,119390.000000,119390.000000,119390.000000,119390.000000,119390.000000,119390.000000,119386.000000,119390.000000,119390.000000,119390.000000,119390.000000,119390.000000,103050.000000,6797.000000,119390.000000,119390.000000,119390.000000,119390.000000
mean,0.370416,104.011416,2016.156554,27.165173,15.798241,0.927599,2.500302,1.856403,0.103890,0.007949,0.031912,0.087118,0.137097,0.221124,86.693382,189.266735,2.321149,101.831122,0.062518,0.571363
std,0.482918,106.863097,0.707476,13.605138,8.780829,0.998613,1.908286,0.579261,0.398561,0.097436,0.175767,0.844336,1.497437,0.652306,110.774548,131.655015,17.594721,50.535790,0.245291,0.792798
min,0.000000,0.000000,2015.000000,1.000000,1.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,1.000000,6.000000,0.000000,-6.380000,0.000000,0.000000
25%,0.000000,18.000000,2016.000000,16.000000,8.000000,0.000000,1.000000,2.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,9.000000,62.000000,0.000000,69.290000,0.000000,0.000000
50%,0.000000,69.000000,2016.000000,28.000000,16.000000,1.000000,2.000000,2.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,14.000000,179.000000,0.000000,94.575000,0.000000,0.000000
75%,1.000000,160.000000,2017.000000,38.000000,23.000000,2.000000,3.000000,2.000000,0.000000,0.000000,0.000000,0.000000,0.000000,0.000000,229.000000,270.000000,0.000000,126.000000,0.000000,1.000000
max,1.000000,737.000000,2017.000000,53.000000,31.000000,19.000000,50.000000,55.000000,10.000000,10.000000,1.000000,26.000000,72.000000,21.000000,535.000000,543.000000,391.000000,5400.000000,8.000000,5.000000


## Kratak zaključak početnog pregleda

Skup podataka sadrži **119390 redova i 36 kolona**, pri čemu svaki red predstavlja jednu hotelsku rezervaciju. Kolone obuhvataju informacije o hotelu, gostima, terminu boravka, načinu rezervisanja, prethodnim rezervacijama i statusu otkazivanja.

Početni pregled pokazuje da skup sadrži i numeričke i kategorijske promenljive. Uočeno je da pojedine numeričke kolone imaju veoma velike maksimalne vrednosti u odnosu na njihove medijane, što može ukazivati na prisustvo netipičnih ili ekstremnih vrednosti. Takođe, broj validnih vrednosti nije isti u svim kolonama, pa je potrebno detaljno proveriti nedostajuće podatke.

U narednim koracima biće analizirane nedostajuće vrednosti, zatim netipične i ekstremne vrednosti, nakon čega će biti urađena detaljnija eksplorativna analiza i ispitivanje odnosa sa ciljnom promenljivom `is_canceled`.

## 4. Analiza nedostajućih vrednosti

Nedostajući podaci mogu biti zapisani kao prazna polja u originalnom CSV fajlu. Međutim, `pandas.read_csv()` ih pri standardnom učitavanju automatski prepoznaje i predstavlja kao `NaN` vrednosti. Zbog toga će se u ovom notebook-u svi nedostajući podaci proveravati pomoću metode `isna()`.

In [31]:
missing_counts = df.isna().sum()
missing_percentages = (missing_counts / len(df) * 100).round(3)

missing_summary = pd.DataFrame({
    "Broj nedostajućih vrednosti": missing_counts,
    "Procenat nedostajućih vrednosti": missing_percentages
})

missing_summary = missing_summary[
    missing_summary["Broj nedostajućih vrednosti"] > 0
].sort_values("Broj nedostajućih vrednosti", ascending=False)

missing_summary

,Broj nedostajućih vrednosti,Procenat nedostajućih vrednosti
company,112593,94.307
agent,16340,13.686
country,488,0.409
children,4,0.003


### Ukupan broj redova sa najmanje jednom nedostajućom vrednošću

Ukupan broj nedostajućih polja predstavlja zbir svih `NaN` vrednosti u dataset-u. Sa druge strane, broj redova sa nedostajućim vrednostima pokazuje koliko rezervacija ima najmanje jedno polje bez podatka.

In [32]:
total_missing_values = df.isna().sum().sum()
rows_with_missing = df.isna().any(axis=1).sum()
rows_with_missing_percentage = (rows_with_missing / len(df) * 100).round(2)

print(f"Ukupan broj nedostajućih polja: {total_missing_values}")
print(f"Broj redova sa najmanje jednom NaN vrednošću: {rows_with_missing}")
print(f"Procenat redova sa najmanje jednom NaN vrednošću: {rows_with_missing_percentage}%")

Ukupan broj nedostajućih polja: 129425
Broj redova sa najmanje jednom NaN vrednošću: 119173
Procenat redova sa najmanje jednom NaN vrednošću: 99.82%


### Zaključak analize nedostajućih vrednosti

Nedostajuće vrednosti pronađene su u kolonama `children`, `country`, `agent` i `company`. Kolona `company` ima ubedljivo najveći procenat nedostajućih vrednosti, dok `agent` ima umeren broj nedostajućih zapisa. Kolone `country` i `children` imaju veoma mali procenat nedostajućih vrednosti.

Visok procenat redova sa najmanje jednom nedostajućom vrednošću najvećim delom je posledica kolone `company`. Zbog toga se ne može zaključiti da je skoro ceo dataset neupotrebljiv, već je potrebno posebno razmotriti značenje nedostajućih vrednosti u svakoj koloni.

U ovom koraku podaci nisu menjani. Način obrade biće određen u kasnijem delu čišćenja, nakon dodatne provere značenja kolona i uticaja nedostajućih vrednosti na dalju analizu.

## 5. Provera dupliranih redova

Proverava se da li u skupu podataka postoje potpuno identični redovi, jer bi njihovo prisustvo moglo da utiče na statističku analizu i modele mašinskog učenja.

In [33]:
duplicate_count = df.duplicated().sum()
duplicate_percentage = round(duplicate_count / len(df) * 100, 2)

print(f"Broj dupliranih redova: {duplicate_count}")
print(f"Procenat dupliranih redova: {duplicate_percentage}%")

Broj dupliranih redova: 0
Procenat dupliranih redova: 0.0%


### Zaključak provere duplikata

U skupu podataka nisu pronađeni potpuno duplirani redovi, pa nije potrebno vršiti njihovo uklanjanje.

## 6. Analiza netipičnih i ekstremnih vrednosti

Osnovna deskriptivna statistika ukazuje na prisustvo vrlo malih, vrlo velikih ili potencijalno nelogičnih vrednosti u pojedinim kolonama. Ovde prvo proveravamo koliko takvih vrednosti ima i u kojim se kolonama pojavljuju, kako bismo kasnije mogli da odlučimo šta sa njima treba uraditi.

### Kreiranje pomoćnih promenljivih za proveru

Privremeno se računaju `total_guests`, kao zbir odraslih, dece i beba, i `total_stay`, kao zbir noćenja tokom vikenda i radnih dana. Ove promenljive se trenutno koriste samo za proveru konzistentnosti podataka. Originalne kolone se ne menjaju i podaci još nisu očišćeni.

In [34]:
df["total_guests"] = (
    df["adults"] + df["children"].fillna(0) + df["babies"]
)

df["total_stay"] = (
    df["stays_in_weekend_nights"] + df["stays_in_week_nights"]
)

### Sistematska provera sumnjivih vrednosti

U nastavku se proveravaju potencijalno nelogične ili veoma retke vrednosti u najvažnijim numeričkim kolonama. Za svaki uslov računa se broj i procenat redova koji ga zadovoljavaju.

In [35]:
unusual_checks = [
    ("Negativna ADR vrednost", df["adr"] < 0),
    ("ADR veći od 1000", df["adr"] > 1000),
    ("Rezervacija bez gostiju", df["total_guests"] == 0),
    ("Rezervacija bez noćenja", df["total_stay"] == 0),
    ("Boravak duži od 30 noći", df["total_stay"] > 30),
    ("Lead time duži od 365 dana", df["lead_time"] > 365),
    ("Više od 10 odraslih gostiju", df["adults"] > 10),
    ("Više od 5 dece", df["children"] > 5),
    ("Više od 5 beba", df["babies"] > 5),
    ("Više od 5 prethodnih otkazivanja", df["previous_cancellations"] > 5),
    ("Više od 20 prethodnih neotkazanih rezervacija", df["previous_bookings_not_canceled"] > 20),
    ("Više od 5 izmena rezervacije", df["booking_changes"] > 5),
    ("Čekanje duže od 100 dana", df["days_in_waiting_list"] > 100),
    ("Više od 3 potrebna parking mesta", df["required_car_parking_spaces"] > 3)
]

unusual_values_summary = pd.DataFrame([
    {
        "Provera": description,
        "Broj redova": condition.sum(),
        "Procenat redova": round(condition.sum() / len(df) * 100, 3)
    }
    for description, condition in unusual_checks
])

unusual_values_summary = unusual_values_summary.sort_values(
    "Broj redova", ascending=False
).reset_index(drop=True)

unusual_values_summary

,Provera,Broj redova,Procenat redova
0,Lead time duži od 365 dana,3148,2.637
1,Čekanje duže od 100 dana,747,0.626
2,Rezervacija bez noćenja,715,0.599
3,Više od 5 prethodnih otkazivanja,202,0.169
4,Rezervacija bez gostiju,180,0.151
5,Više od 5 izmena rezervacije,149,0.125
6,Više od 20 prethodnih neotkazanih rezervacija,135,0.113
7,Boravak duži od 30 noći,24,0.020
8,Više od 10 odraslih gostiju,12,0.010
9,Više od 5 beba,2,0.002


Isti zapis može istovremeno pripadati većem broju navedenih grupa, na primer rezervacija može biti i bez gostiju i bez noćenja. Zbog toga se brojevi iz tabele ne sabiraju kao broj različitih problematičnih redova.

### Pregled konkretnih primera

Prikazani su primeri najvažnijih sumnjivih vrednosti radi njihovog detaljnijeg pregleda i tumačenja. U svakom prikazu izdvojene su relevantne kolone, a najekstremnije vrednosti prikazane su prve.

In [36]:
adr_extreme_rows = df[(df["adr"] < 0) | (df["adr"] > 1000)]
adr_columns = [
    "hotel", "is_canceled", "adr", "adults", "children",
    "babies", "total_stay", "reservation_status"
]

if adr_extreme_rows.empty:
    print("Nisu pronađeni redovi sa negativnim ADR-om ili ADR-om većim od 1000.")
else:
    display(
        adr_extreme_rows.sort_values(
            "adr", key=lambda values: values.abs(), ascending=False
        )[adr_columns].head(10)
    )

,hotel,is_canceled,adr,adults,children,babies,total_stay,reservation_status
48515,City Hotel,1,5400.00,2,0.0,0,1,Canceled
14969,Resort Hotel,0,-6.38,2,0.0,0,10,Check-Out


U koloni adr izdvajaju se dve vrednosti: jedna negativna i jedna izrazito visoka. Pošto dataset ne sadrži dodatno objašnjenje za ove zapise, nije moguće pouzdano utvrditi da li su u pitanju greške ili validni slučajevi. Zato će se kasnije proveriti gde se nalaze u raspodeli i koliko utiču na prosek i medijanu.

In [42]:
zero_guest_rows = df[df["total_guests"] == 0]
guest_columns = [
    "hotel", "is_canceled", "adults", "children", "babies",
    "total_guests", "total_stay", "adr", "reservation_status"
]

if zero_guest_rows.empty:
    print("Nisu pronađene rezervacije bez gostiju.")
else:
    display(
        zero_guest_rows.sort_values(
            ["total_stay", "adr"], ascending=False
        )[guest_columns].head(10)
    )

,hotel,is_canceled,adults,children,babies,total_guests,total_stay,adr,reservation_status
101794,City Hotel,0,0,0.0,0,0.0,57,8.34,Check-Out
88017,City Hotel,0,0,0.0,0,0.0,49,0.00,Check-Out
106561,City Hotel,0,0,0.0,0,0.0,43,0.00,Check-Out
41020,City Hotel,0,0,0.0,0,0.0,29,0.00,Check-Out
98046,City Hotel,0,0,0.0,0,0.0,28,0.00,Check-Out
109038,City Hotel,0,0,0.0,0,0.0,27,0.00,Check-Out
100563,City Hotel,0,0,0.0,0,0.0,22,0.00,Check-Out
106562,City Hotel,0,0,0.0,0,0.0,15,35.81,Check-Out
60651,City Hotel,1,0,0.0,0,0.0,15,0.00,Canceled
93204,City Hotel,0,0,0.0,0,0.0,15,0.00,Check-Out


Pronađene su i rezervacije u kojima nije evidentiran nijedan gost. Takvi zapisi deluju neuobičajeno, pa će se pogledati njihov status, broj noćenja i ADR kako bi se bolje razumelo o kakvim rezervacijama je reč i da li ih treba zadržati, izmeniti ili ukloniti.

### Rezervacije bez noćenja

To su rezervacije kod kojih je zbir kolona `stays_in_weekend_nights` i `stays_in_week_nights` jednak nuli. Pre donošenja odluke o njihovoj obradi potrebno je proveriti status rezervacije, broj gostiju i ADR.

In [38]:
zero_stay_rows = df[df["total_stay"] == 0]
zero_stay_columns = [
    "hotel", "is_canceled", "stays_in_weekend_nights",
    "stays_in_week_nights", "total_stay", "total_guests",
    "adr", "reservation_status", "reservation_status_date"
]

if zero_stay_rows.empty:
    print("Nisu pronađene rezervacije bez noćenja.")
else:
    display(
        zero_stay_rows.sort_values(
            ["is_canceled", "adr"], ascending=[False, False]
        )[zero_stay_columns].head(10)
    )

,hotel,is_canceled,stays_in_weekend_nights,stays_in_week_nights,total_stay,total_guests,adr,reservation_status,reservation_status_date
4127,Resort Hotel,1,0,0,0,0.0,0.0,Canceled,2016-02-15
9376,Resort Hotel,1,0,0,0,0.0,0.0,Canceled,2016-11-21
10101,Resort Hotel,1,0,0,0,1.0,0.0,Canceled,2017-02-12
19503,Resort Hotel,1,0,0,0,2.0,0.0,No-Show,2015-12-23
19823,Resort Hotel,1,0,0,0,2.0,0.0,No-Show,2016-01-04
22557,Resort Hotel,1,0,0,0,1.0,0.0,Canceled,2016-03-28
22970,Resort Hotel,1,0,0,0,1.0,0.0,Canceled,2016-04-12
23734,Resort Hotel,1,0,0,0,2.0,0.0,No-Show,2016-04-30
24660,Resort Hotel,1,0,0,0,2.0,0.0,Canceled,2016-05-29
25333,Resort Hotel,1,0,0,0,1.0,0.0,No-Show,2016-06-21


Rezervacije bez noćenja ne mogu se automatski smatrati greškom. Njihovo značenje zavisi od statusa rezervacije, pa će se proveriti koliko među njima ima otkazanih, realizovanih i `No-Show` rezervacija, kao i da li imaju evidentirane goste i ADR vrednost.

In [39]:
guest_extreme_rows = df[
    (df["adults"] > 10)
    | (df["children"] > 5)
    | (df["babies"] > 5)
]
guest_extreme_columns = [
    "hotel", "is_canceled", "adults", "children", "babies",
    "total_guests", "total_stay", "adr"
]

if guest_extreme_rows.empty:
    print("Nisu pronađene rezervacije sa veoma velikim brojem gostiju.")
else:
    display(
        guest_extreme_rows.sort_values(
            "total_guests", ascending=False
        )[guest_extreme_columns]
    )

,hotel,is_canceled,adults,children,babies,total_guests,total_stay,adr
2173,Resort Hotel,1,55,0.0,0,55.0,2,0.00
1643,Resort Hotel,1,50,0.0,0,50.0,3,0.00
1539,Resort Hotel,1,40,0.0,0,40.0,3,0.00
1962,Resort Hotel,1,27,0.0,0,27.0,4,0.00
1917,Resort Hotel,1,27,0.0,0,27.0,4,0.00
1752,Resort Hotel,1,26,0.0,0,26.0,7,0.00
1587,Resort Hotel,1,26,0.0,0,26.0,7,0.00
2003,Resort Hotel,1,26,0.0,0,26.0,7,0.00
1884,Resort Hotel,1,26,0.0,0,26.0,7,0.00
2164,Resort Hotel,1,26,0.0,0,26.0,7,0.00


U malom broju rezervacija evidentiran je neuobičajeno veliki broj gostiju. Pošto takvi slučajevi mogu predstavljati i validne grupne rezervacije, nije ih moguće automatski smatrati greškom. Zbog toga će se prilikom čišćenja podataka uzeti u obzir i ostale informacije iz rezervacije kako bi se donela odgovarajuća odluka.

In [40]:
long_lead_time_rows = df[df["lead_time"] > 365]
long_waiting_rows = df[df["days_in_waiting_list"] > 100]

time_columns = [
    "hotel",
    "is_canceled",
    "lead_time",
    "days_in_waiting_list",
    "deposit_type",
    "market_segment",
    "reservation_status"
]

print("Primeri rezervacija sa lead_time dužim od 365 dana:")

if long_lead_time_rows.empty:
    print("Nisu pronađene takve rezervacije.")
else:
    display(
        long_lead_time_rows
        .sort_values("lead_time", ascending=False)
        [time_columns]
        .head(5)
    )

print("\nPrimeri rezervacija sa čekanjem dužim od 100 dana:")

if long_waiting_rows.empty:
    print("Nisu pronađene takve rezervacije.")
else:
    display(
        long_waiting_rows
        .sort_values("days_in_waiting_list", ascending=False)
        [time_columns]
        .head(5)
    )

Primeri rezervacija sa lead_time dužim od 365 dana:


,hotel,is_canceled,lead_time,days_in_waiting_list,deposit_type,market_segment,reservation_status
1,Resort Hotel,0,737,0,No Deposit,Direct,Check-Out
4182,Resort Hotel,0,709,0,No Deposit,Direct,Check-Out
65245,City Hotel,1,629,0,Non Refund,Groups,Canceled
65253,City Hotel,1,629,0,Non Refund,Groups,Canceled
65240,City Hotel,1,629,0,Non Refund,Groups,Canceled



Primeri rezervacija sa čekanjem dužim od 100 dana:


,hotel,is_canceled,lead_time,days_in_waiting_list,deposit_type,market_segment,reservation_status
59454,City Hotel,1,464,391,Non Refund,Groups,Canceled
59450,City Hotel,1,464,391,Non Refund,Groups,Canceled
59444,City Hotel,1,464,391,Non Refund,Groups,Canceled
59434,City Hotel,1,464,391,Non Refund,Groups,Canceled
59435,City Hotel,1,464,391,Non Refund,Groups,Canceled


Veliki `lead_time` i dugo zadržavanje na listi čekanja predstavljaju retke, ali moguće vrednosti. One se za sada ne smatraju greškama. Kasnije će se analizirati njihova raspodela i povezanost sa ciljnom promenljivom `is_canceled`.

### Zaključak analize netipičnih i ekstremnih vrednosti

Analizom su pronađene potencijalno nelogične vrednosti, kao što su negativan ADR, rezervacije bez evidentiranih gostiju i rezervacije bez noćenja. Uočene su i retke, ali moguće vrednosti u kolonama `lead_time`, `days_in_waiting_list`, broju gostiju i trajanju boravka.

U ovom delu nijedan zapis nije uklonjen niti izmenjen. Sledeći korak biće ciljana analiza pronađenih slučajeva: proveriće se njihova raspodela, status rezervacije, odnos sa ciljnom promenljivom `is_canceled` i uticaj na osnovne statističke pokazatelje. Na osnovu tih rezultata biće donete i obrazložene odluke o tome koje vrednosti ostaju, koje se uklanjaju i koje se obrađuju na drugi način.